src/PrestaShopBundle/Form/Admin/Type/VisibilityChoiceType.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Class CountryChoiceType is responsible for providing country choices with -- symbol in front of array.
+ */
+class VisibilityChoiceType extends AbstractType
+{
```

Comment:
This probably needs a better name. So far it only adds visibility param to choice type but may be it will be alerted in the future, but I am not sure what to call it in that case . CustomChoiceType, ImprovedChoiceType?  Neither seems to be an optimal name.

Reply:
So this is a form input that can be visible OR invisible due to other parts of the form, is that right ? Like "France" has no state but "USA" has states so if I choose France of USA in the "country" dropdown, this toggles the "states" form.

All of this "toggle input on country input change" behavior is already handled by JS I think ?

If yes, I see 2 options to address your question/issue:
- either we consider it's some extra data but it's not worth it creating a new php class, so we pass it as extra data or as attribute
- either we choose to create a php class ; then maybe ToggleableChoiceType ? it says what it is: "this choice type can be toggled"

Reply:
I added it as attribute as I think it's cleaner this way. But there is one reason I dislike it: it's not clear that you can just add visible => false to attributes and make whole block invisible. However there is no way to know this other then look up the the prestashop_ui_kit.html.twig file or find example where it has been used. 

Any way we can now indicate what options are available in attr parameter?

Reply:
> Any way we can indicate what options are available in attr parameter?

None than I know. But I think code is straightforward here, it's not an issue.